docs/sharding.md

```diff
@@ -101,7 +101,98 @@ namespace SimpleChatV3
 }
 ```
 
-For other languages such as JavaScript, we will support route algorithm customization in the future.
+> The dependency of customized negotiation router: 
+> 
+> If you need to implement [`GetNegotiateEndpoint(HttpContext context, IEnumerable<ServiceEndpoint> endpoints)`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR/EndpointRouters/IEndpointRouter.cs) method yourself and rely on the parameter `HttpContext`, for example, use `HttpContext.Request.Query["endpoint"]` to select a nearer endpoint for router, you can only use the HTTP trigger in your negotiation functions, so that your router can get the `HttpContext` object correctly.
```

Comment:
`GetNegotiateEndpoint` comes from nowhere... how about add a brief introduction about the interface `IEndpointRouter`? Not clear here, maybe add a code sample for using httpcontext can be better